horton/gbasis/ints.h

```diff
@@ -126,41 +126,315 @@ class GB2MomentIntegral: public GB2Integral {
+  double ab[3];                 //!< Relative vector from shell 2 to 0 (LibInt order).
+  double cd[3];                 //!< Relative vector from shell 3 to 1 (LibInt order).
+  double ab2;                   //!< Norm squared of ab.
+  double cd2;                   //!< Norm squared of cd.
+};
+
+
+/** @brief
+        Electron repulsion four-center integrals.
+
+    The potential is 1/r.
+  */
+class GB4ElectronRepulsionIntegralLibInt : public GB4IntegralLibInt {
+ public:
+  /** @brief
+          Initialize a GB4ElectronRepulsionIntegralLibInt object.
+
+      @param max_shell_type
+          Highest angular momentum index to be expected in the reset method.
+    */
+  explicit GB4ElectronRepulsionIntegralLibInt(long max_shell_type)
+      : GB4IntegralLibInt(max_shell_type) {}
+
+  /** @brief
+          Evaluate the Laplace transform of the ordinary Coulomb potential.
+
+      See Eq. (39) in Ahlrichs' paper. This is basically a rescaled Boys function.
+
+      See base class for more details.
+    */
+  virtual void laplace_of_potential(double prefac, double rho, double t, long mmax,
+                                    double* output);
+};
+
+
+/** @brief
+        Short-range electron repulsion four-center integrals.
+
+    The potential is erf(mu*r)/r.
+  */
+class GB4ErfIntegralLibInt : public GB4IntegralLibInt {
+ public:
+  /** @brief
+          Initialize a GB4ErfIntegralLibInt object.
+
+      @param max_shell_type
+          Highest angular momentum index to be expected in the reset method.
+
+      @param mu
+          The range-separation parameter
+    */
+  GB4ErfIntegralLibInt(long max_shell_type, double mu)
+      : GB4IntegralLibInt(max_shell_type), mu(mu) {}
+
+  /** @brief
+          Evaluate the Laplace transform of the long-range Coulomb potential.
+          (The short-range part is damped away using an error function.) See (52) in
+          Ahlrichs' paper.
+
+      See base class for more details.
+    */
+  virtual void laplace_of_potential(double prefac, double rho, double t, long mmax,
+                                    double* output);
+
+  const double get_mu() const {return mu;}  //!< The range-separation parameter.
+
+ private:
+  double mu;  //!< The range-separation parameter.
+};
+
+
+/** @brief
+        Gaussian electron repulsion four-center integrals.
+
+    The potential is ??? TODO
+  */
+class GB4GaussIntegralLibInt : public GB4IntegralLibInt {
+ public:
+  /** @brief
+          Initialize a GB4GaussIntegralLibInt object.
+
+      @param max_shell_type
+          Highest angular momentum index to be expected in the reset method.
+
+      @param c
+          Coefficient of the gaussian.
+
+      @param alpha
+          Exponential parameter of the gaussian.
+    */
+  GB4GaussIntegralLibInt(long max_shell_type, double c, double alpha)
+      : GB4IntegralLibInt(max_shell_type), c(c), alpha(alpha) {}
+
+  /** @brief
+          Evaluate the Laplace transform of the Gaussian potential. Reference? TODO
```

Comment:
I just checked, Julien and Savin explained the Rys quadrature scheme in their paper, so I guess there is no published reference (Long-range–short-range separation of the electron-electron interaction
in density-functional theory, PHYSICAL REVIEW A 70, 062505 (2004)). For the Obara-Saika scheme I have my own notes, based on their implementation and Alrichs paper:
\begin{equation}
g(r) = c e^{- \alpha^2 r^2}: G_0 = (\frac{\pi}{\rho + \alpha})^{3/2} c e^{-T f(\alpha)}
\end{equation}
where:
\begin{equation}
f(\alpha) = \frac{\alpha}{\rho+\alpha}.
\end{equation}
The derivatives are:
\begin{equation}
G_n = (\frac{\pi}{\rho + \alpha})^{3/2} f(\alpha)^n c e^{-T f(\alpha)}
\end{equation}.

I attached also my notes. 
[GaussIntegrals.pdf](https://github.com/theochem/horton/files/365174/GaussIntegrals.pdf)